src/chapters/unit-testing/mstest.rst

```diff
@@ -0,0 +1,303 @@
+   :alt: WINDOWS: Visual Studio open Test Explorer
+
+   WINDOWS: Visual Studio open Test Explorer
+
+With the panel open, select the *Run All Tests* option.
+
+.. admonition:: Note
+
+   If you see that the test fails to run, neither passing nor failing, you may need to adjust a setting to use
+   64bit processing.
+
+   .. figure:: ./figures/vs-windows-process-architecture-setting.png
+      :alt: WINDOWS: Set Test Explorer to use x64 process
+
+      WINDOWS: Set Test Explorer to use x64 process
+
+   You may also need to update some of the testing packages. Right click on the 
+   ``CarTests`` project and select *Manage NuGet Packages...*. If you see some items
+   in the *Update* section of the panel that opens, run the updates. Close and reopen 
+   the *Team Explorer* panel and *Visual Studio* to ensure the changes are applied.
+
+All Users: Output and Adding More Tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once run, you'll see a new output panel with a green check mark indicating the test passed and a message 
+stating the test passed. 
+
+We know now how the test runner behaves when a test passes and can begin the real work of unit 
+testing the ``Car`` class. To test the ``Car`` class, we must make it available to us by adding 
+``using CarNS;`` to the top of your file.
+
+Under your second TODO, write a test to verify that the constructor sets the 
+``gasTankLevel`` field.
+
+.. sourcecode:: c#
+   :lineno-start: 16
+
+   //TODO: constructor sets gasTankLevel properly
+   [TestMethod]
+   public void TestInitialGasTank()
+   {
+      Car test_car = new Car("Toyota", "Prius", 10, 50);
+      Assert.AreEqual(10, test_car.GasTankLevel, .001);
+   }
+
+Here, we give the test a descriptive name, ``TestInitialGasTank()``, initialize a new 
+``Car`` object, and test that the constructor correctly sets the ``gasTankLevel`` field.
+
+Run ``CarTest`` to see that both tests pass. 
+
+.. admonition:: Tip
+
+   If you want to rerun only one test, right click on its listing in the results pane.
+
+
+.. index:: ! [TestInitialize]
+
+``[TestInitialize]``
+--------------------
+
+While ``[TestClass]`` and ``[TestMethod]`` are required to run tests, there are many other 
+attributes you may find useful as your test files grow in scope. One such item to know
+is **[TestInitialize]**. ``[TestInitialize]`` will run before each test method is run in a class. 
+
+In the case of ``CarTest``, it would be nice to not need to create a new ``Car`` instance for 
+each test we write. In your ``TestInitialGasTank()`` method, remove the line initiating ``test_car``. 
+Above your relevant test, add the following ``[TestInitialize]`` method:
+
+.. sourcecode:: c#
+   :lineno-start: 16
+
+   Car test_car;
+
+   [TestInitialize]
+   public void CreateCarObject()
+   {
+      test_car = new Car("Toyota", "Prius", 10, 50);
+   }
+
+Now, run the test project and ensure your test still passes.
+
+.. index:: ! [TestCleanup]
+
+``[TestCleanup]``
+-----------------
```

Comment:
I would provide a link for students who want to learn more about the use cases of this attribute.